test/dynamo/test_compile.py

```diff
@@ -99,3 +100,56 @@ def test_dynamo_signatures(self):
 
         for fn_name in function_names:
             self.check_signature(fn_name, fn_name, torch._dynamo)
+
+
+class ResumeAfterRegionTests(DynamoTestCase):
+    def test_resume_at_after_try_except(self):
+        def fn(x):
+            x = x + 1
+            # Should fallback on eager here
+            try:
+                x += 3
+                raise AssertionError()
+            except AssertionError:
+                pass
+            # Should resume at here
+            x = x + 2
+            return x
+
+        x = torch.zeros(3)
+        eager = fn(x)
+
+        cnt = CompileCounter()
+        compiled_fn = torch._dynamo.optimize(backend=cnt)(fn)
+        compiled = compiled_fn(x)
+
+        self.assertEqual(eager, compiled)
+        self.assertEqual(cnt.frame_count, 2)
+
+    def test_resume_at_after_try_except_nested(self):
```

Comment:
Please add test that correct behaviour occurs when nesting try except within a with context manager